rust-runtime/aws-smithy-http/src/body.rs

```diff
@@ -323,13 +264,47 @@ impl http_body::Body for SdkBody {
         match &self.inner {
             Inner::Once { inner: None } => SizeHint::with_exact(0),
             Inner::Once { inner: Some(bytes) } => SizeHint::with_exact(bytes.len() as u64),
-            Inner::Streaming { inner: hyper_body } => hyper_body.size_hint(),
-            Inner::Dyn { inner: box_body } => box_body.size_hint(),
+            Inner::Streaming { inner } => http_body::Body::size_hint(inner),
+            Inner::Dyn { inner } => http_body::Body::size_hint(inner),
             Inner::Taken => SizeHint::new(),
         }
     }
 }
 
+const SIZE_HINT_32_BIT_PANIC_MESSAGE: &str = r#"
+You're running a 32-bit system and this stream's length is too large to be represented with a usize.
+Please limit stream length to less than 4.294Gb or run this program on a 64-bit computer architecture.
+"#;
+
+impl Stream for SdkBody {
```

Comment:
The reasoning for not doing this up until now was that [`AsyncIterator`](https://doc.rust-lang.org/std/async_iter/trait.AsyncIterator.html) was close to becoming stabilized in `std`, and this will be obsolete as soon as that happens. This is definitely worth discussing further since it looks like that will take longer due to discussion of generic keywords. Maybe we should create another runtime crate similar to `aws-smithy-types-convert` that adds in useful conversions to the current set of async traits that are widely used until they become `std`.

There are a couple of crates that need to get revisited as we get closer to a stable release:
- https://github.com/awslabs/smithy-rs/blob/cbbbec4a4892e301ebde3af5cc852aa48d82e608/rust-runtime/aws-smithy-async/external-types.toml#L2
- https://github.com/awslabs/smithy-rs/blob/cbbbec4a4892e301ebde3af5cc852aa48d82e608/rust-runtime/aws-smithy-http/external-types.toml#L23